src/hotspot/cpu/aarch64/templateTable_aarch64.cpp

```diff
@@ -2320,13 +2320,83 @@ void TemplateTable::load_field_cp_cache_entry(Register obj,
   }
 }
 
+void TemplateTable::load_invokedynamic_entry(Register method) {
+  // setup registers
+  const Register appendix = r0;
+  const Register cache = r2;
+  const Register index = r3;
+  assert_different_registers(method, appendix, cache, index, rcpool);
+
+  __ save_bcp();
+
+  Label resolved;
+
+  __ load_resolved_indy_entry(cache, index);
+  // Load-acquire the adapter method
+  __ lea(method, Address(cache, in_bytes(ResolvedIndyEntry::method_offset())));
+  __ ldar(method, method);
```

Comment:
What reordering are we trying to prevent here?

Reply:
The loads of the data stored in `ResolvedIndyEntry::fill_in()` must not be reordered with loading  `ResolvedIndyEntry::_method`.

Reply:
Ah, I see. This acquiring load matches the releasing store in `fill_in()`. Please add a comment here to that effect.